### ISSUE TITLE
Intermittent UI test failure - org.mozilla.fenix.ui.HistoryTest - visitedUrlHistoryTest

### ISSUE THREAD

**Issue by Archaeopteryx** [mozilla-mobile/fenix#22304]
### Firebase Test Run:
https://console.firebase.google.com/project/moz-fenix/testlab/histories/bh.80a1a071cfb2770e/matrices/7787466636091778085/executions/bs.cc6aa8a48f33c22a/testcases/1
### Stacktrace:
```
androidx.test.espresso.NoMatchingViewException: No views in hierarchy found matching: (with res-name that is "browserLayout" or with res-name that is "download_button")

View Hierarchy:
+>DecorView{id=-1, visibility=VISIBLE, width=1080, height=1920, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params={(0,0)(fillxfill) sim={adjust=resize} ty=BASE_APPLICATION wanim=0x7f14034b
  fl=LAYOUT_IN_SCREEN LAYOUT_INSET_DECOR HARDWARE_ACCELERATED DRAWS_SYSTEM_BAR_BACKGROUNDS
  pfl=FORCE_DRAW_STATUS_BAR_BACKGROUND
  vsysui=LIGHT_STATUS_BAR LIGHT_NAVIGATION_BAR}, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=2} 
|
+->LinearLayout{id=-1, visibility=VISIBLE, width=1080, height=1788, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@6b2b71, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=2} 
|
+-->ViewStub{id=16908682, res-name=action_mode_bar_stub, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@f441d56, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0} 
|
+-->FrameLayout{id=-1, visibility=VISIBLE, width=1080, height=1722, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@774e1d7, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=66.0, child-count=1} 
|
+--->FitWindowsLinearLayout{id=2131361870, res-name=action_bar_root, visibility=VISIBLE, width=1080, height=1722, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@ab60dad, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=2} 
|
+---->ViewStubCompat{id=2131361938, res-name=action_mode_bar_stub, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@87cb8e2, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0} 
|
+---->ContentFrameLayout{id=16908290, res-name=content, visibility=VISIBLE, width=1080, height=1722, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@e3bc473, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=1} 
|
+----->RootLinearLayout{id=2131362806, res-name=rootContainer, visibility=VISIBLE, width=1080, height=1722, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@552d7a9, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=2} 
|
+------>ViewStub{id=2131362652, res-name=navigationToolbarStub, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@7b1812e, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0} 
|
+------>FragmentContainerView{id=2131362208, res-name=container, visibility=VISIBLE, width=1080, height=1722, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.LinearLayout$LayoutParams@38434cf, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=1} 
|
+------->CoordinatorLayout{id=2131362448, res-name=homeLayout, visibility=VISIBLE, width=1080, height=1722, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@e7bbe5c, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=3} 
|
+-------->AppBarLayout{id=2131362445, res-name=homeAppBar, visibility=VISIBLE, width=1080, height=248, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.coordinatorlayout.widget.CoordinatorLayout$LayoutParams@5d84565, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=1} 
|
+--------->CollapsingToolbarLayout{id=-1, visibility=VISIBLE, width=1080, height=248, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=com.google.android.material.appbar.AppBarLayout$LayoutParams@358823a, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, child-count=2} 
|
+---------->AppCompatImageButton{id=2131362743, res-name=privateBrowsingButton, desc=Enable private browsing, visibility=VISIBLE, width=132, height=132, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=com.google.android.material.appbar.CollapsingToolbarLayout$LayoutParams@7394eeb, tag=null, root-is-layout-requested=false, has-input-connection=false, x=915.0, y=39.0} 
|
+---------->AppCompatImageView{id=2131363157, res-name=wordmark, desc=Firefox Nightly, visibility=VISIBLE, width=595, height=110, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=com.google.android.material.appbar.CollapsingToolbarLayout$LayoutParams@e346448, tag=null, root-is-layout-requested=false, has-input-connection=false, x=44.0, y=50.0} 
|
+-------->RecyclerView{id=2131362883, res-name=sessionControlRecyclerView, visibility=VISIBLE, width=1080, height=1479, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.coordinatorlayout.widget.CoordinatorLayout$LayoutParams@a8cd2e1, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=248.0, child-count=3} 
|
+--------->FrameLayout{id=2131362687, res-name=onboarding_header, visibility=VISIBLE, width=992, height=78, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.recyclerview.widget.RecyclerView$LayoutParams@13c8806, tag=null, root-is-layout-requested=false, has-input-connection=false, x=44.0, y=44.0, child-count=1} 
|
+---------->AppCompatTextView{id=2131362427, res-name=header_text, visibility=VISIBLE, width=992, height=78, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@2dbeec7, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0, text=Welcome to Firefox Nightly!, input-type=0, ime-target=false, has-links=false} 
|
+--------->ConstraintLayout{id=2131362686, res-name=onboarding_card, visibility=VISIBLE, width=992, height=831, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.recyclerview.widget.RecyclerView$LayoutParams@f4f7cf4, tag=null, root-is-layout-requested=false, has-input-connection=false, x=44.0, y=144.0, child-count=11} 
|
+---------->AppCompatTextView{id=2131362427, res-name=header_text, visibility=VISIBLE, width=548, height=103, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@5a2bc1d, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=44.0, text=Choose your theme, input-type=0, ime-target=false, has-links=false} 
|
+---------->AppCompatTextView{id=2131362283, res-name=description_text, visibility=VISIBLE, width=992, height=99, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@7091e92, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=147.0, text=Save some battery and your eyesight with dark mode., input-type=0, ime-target=false, has-links=false} 
|
+---------->AppCompatTextView{id=2131362505, res-name=light_theme_description, visibility=VISIBLE, width=232, height=59, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@d1eb063, tag=null, root-is-layout-requested=false, has-input-connection=false, x=152.0, y=533.0, text=Light theme, input-type=0, ime-target=false, has-links=false} 
|
+---------->OnboardingRadioButton{id=2131363049, res-name=theme_light_radio_button, desc=Light theme, visibility=VISIBLE, width=88, height=88, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@f617460, tag=null, root-is-layout-requested=false, has-input-connection=false, x=61.70625, y=267.95, text=, input-type=0, ime-target=false, has-links=false, is-checked=false} 
|
+---------->AppCompatImageView{id=2131363048, res-name=theme_light_image, desc=Light theme, visibility=VISIBLE, width=375, height=221, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@33fd19, tag=null, root-is-layout-requested=false, has-input-connection=false, x=81.0, y=290.0} 
|
+---------->OnboardingRadioButton{id=2131363047, res-name=theme_dark_radio_button, desc=Dark theme, visibility=VISIBLE, width=88, height=88, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@79691de, tag=null, root-is-layout-requested=false, has-input-connection=false, x=516.70624, y=267.95, text=, input-type=0, ime-target=false, has-links=false, is-checked=false} 
|
+---------->AppCompatImageView{id=2131363046, res-name=theme_dark_image, desc=Dark theme, visibility=VISIBLE, width=375, height=221, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@b41efbf, tag=null, root-is-layout-requested=false, has-input-connection=false, x=536.0, y=290.0} 
|
+---------->AppCompatTextView{id=2131362255, res-name=dark_theme_title, visibility=VISIBLE, width=224, height=59, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@ed9768c, tag=null, root-is-layout-requested=false, has-input-connection=false, x=612.0, y=533.0, text=Dark theme, input-type=0, ime-target=false, has-links=false} 
|
+---------->View{id=2131362309, res-name=divider, visibility=VISIBLE, width=986, height=3, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@fe51d5, tag=null, root-is-layout-requested=false, has-input-connection=false, x=3.0, y=636.0} 
|
+---------->View{id=2131362170, res-name=clickable_region_automatic, desc=Automatic Adapts to your device settings, visibility=VISIBLE, width=992, height=195, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@8f5edea, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=636.0} 
|
+---------->OnboardingRadioButton{id=2131363045, res-name=theme_automatic_radio_button, desc=Automatic, visibility=VISIBLE, width=617, height=104, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@3ed6f78, tag=null, root-is-layout-requested=false, has-input-connection=false, x=44.0, y=683.0, text=Automatic
Adapts to your device settings, input-type=0, ime-target=false, has-links=false, is-checked=true} 
|
+--------->ConstraintLayout{id=2131362686, res-name=onboarding_card, visibility=VISIBLE, width=992, height=636, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.recyclerview.widget.RecyclerView$LayoutParams@2df3651, tag=null, root-is-layout-requested=false, has-input-connection=false, x=44.0, y=1019.0, child-count=8} 
|
+---------->AppCompatTextView{id=2131362427, res-name=header_text, visibility=VISIBLE, width=753, height=103, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@f3cfeb6, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=44.0, text=Pick your toolbar placement, input-type=0, ime-target=false, has-links=false} 
|
+---------->AppCompatTextView{id=2131362283, res-name=description_text, visibility=VISIBLE, width=992, height=99, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@95817b7, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=147.0, text=Put the toolbar within easy reach. Keep it on the bottom, or move it to the top., input-type=0, ime-target=false, has-links=false} 
|
+---------->OnboardingRadioButton{id=2131363068, res-name=toolbar_bottom_radio_button, desc=Bottom, visibility=VISIBLE, width=88, height=88, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@460b24, tag=null, root-is-layout-requested=false, has-input-connection=false, x=61.70625, y=267.95, text=, input-type=0, ime-target=false, has-links=false, is-checked=true} 
|
+---------->AppCompatImageView{id=2131363067, res-name=toolbar_bottom_image, desc=Bottom, visibility=VISIBLE, width=375, height=221, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=true, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@70fe68d, tag=null, root-is-layout-requested=false, has-input-connection=false, x=81.0, y=290.0} 
|
+---------->AppCompatTextView{id=2131363069, res-name=toolbar_bottom_title, visibility=VISIBLE, width=144, height=59, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@8125042, tag=null, root-is-layout-requested=false, has-input-connection=false, x=196.0, y=533.0, text=Bottom, input-type=0, ime-target=false, has-links=false} 
|
+---------->AppCompatTextView{id=2131363075, res-name=toolbar_top_title, visibility=VISIBLE, width=74, height=59, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@94c7853, tag=null, root-is-layout-requested=false, has-input-connection=false, x=731.0, y=533.0, text=Top, input-type=0, ime-target=false, has-links=false} 
|
+---------->OnboardingRadioButton{id=2131363074, res-name=toolbar_top_radio_button, desc=Top, visibility=VISIBLE, width=88, height=88, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@56ab590, tag=null, root-is-layout-requested=false, has-input-connection=false, x=560.70624, y=267.95, text=, input-type=0, ime-target=false, has-links=false, is-checked=false} 
|
+---------->AppCompatImageView{id=2131363073, res-name=toolbar_top_image, desc=Top, visibility=VISIBLE, width=375, height=221, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@f215e89, tag=null, root-is-layout-requested=false, has-input-connection=false, x=580.0, y=290.0} 
|
+-------->ConstraintLayout{id=2131363066, res-name=toolbarLayout, visibility=VISIBLE, width=1080, height=157, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.coordinatorlayout.widget.CoordinatorLayout$LayoutParams@2f92e8e, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=1565.0, child-count=6} 
|
+--------->View{id=2131362100, res-name=bottom_bar, visibility=VISIBLE, width=1080, height=154, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@1c46af, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=3.0} 
|
+--------->FrameLayout{id=2131363076, res-name=toolbar_wrapper, desc=Search or enter address, visibility=VISIBLE, width=827, height=110, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@3ed9abc, tag=null, root-is-layout-requested=false, has-input-connection=false, x=22.0, y=25.0, child-count=2} 
|
+---------->AppCompatImageView{id=2131362849, res-name=search_engine_icon, visibility=VISIBLE, width=66, height=66, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@ab85a45, tag=null, root-is-layout-requested=false, has-input-connection=false, x=22.0, y=22.0} 
|
+---------->AppCompatTextView{id=2131363065, res-name=toolbar, visibility=VISIBLE, width=433, height=56, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@25da59a, tag=null, root-is-layout-requested=false, has-input-connection=false, x=121.0, y=27.0, text=Search or enter address, input-type=0, ime-target=false, has-links=false} 
|
+--------->Barrier{id=2131361853, res-name=accessory_button_barrier, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@9709ecb, tag=null, root-is-layout-requested=false, has-input-connection=false, x=849.0, y=0.0} 
|
+--------->TabCounter{id=2131362993, res-name=tab_button, visibility=VISIBLE, width=132, height=132, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@157a6a8, tag=null, root-is-layout-requested=false, has-input-connection=false, x=849.0, y=14.0, child-count=1} 
|
+---------->FrameLayout{id=2131362219, res-name=counter_root, desc=1 open tab. Tap to switch tabs., visibility=VISIBLE, width=66, height=66, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.RelativeLayout$LayoutParams@50955c1, tag=null, root-is-layout-requested=false, has-input-connection=false, x=33.0, y=33.0, child-count=2} 
|
+----------->AppCompatImageView{id=2131362218, res-name=counter_box, desc=The tab counter toolbar button., visibility=VISIBLE, width=66, height=66, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@1608166, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0} 
|
+----------->AppCompatTextView{id=2131362220, res-name=counter_text, visibility=VISIBLE, width=19, height=44, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@8685ca7, tag=null, root-is-layout-requested=false, has-input-connection=false, x=23.0, y=10.0, text=1, input-type=0, ime-target=false, has-links=false} 
|
+--------->MenuButton{id=2131362539, res-name=menuButton, visibility=VISIBLE, width=99, height=132, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@e148cfd, tag=null, root-is-layout-requested=false, has-input-connection=false, x=981.0, y=14.0, child-count=3} 
|
+---------->AppCompatImageView{id=2131362431, res-name=highlight, desc=Highlighted, visibility=GONE, width=0, height=0, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=true, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@a634df2, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0} 
|
+---------->AppCompatImageView{id=2131362464, res-name=icon, desc=Menu, visibility=VISIBLE, width=99, height=132, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@9341c43, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0} 
|
+---------->AppCompatImageView{id=2131362681, res-name=notification_dot, desc=Highlighted, visibility=VISIBLE, width=99, height=132, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@a9ea2c0, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0} 
|
+--------->View{id=2131362099, res-name=bottomBarShadow, visibility=VISIBLE, width=1080, height=3, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@ba1fbf9, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=0.0} 
|
+->View{id=16908336, res-name=navigationBarBackground, visibility=VISIBLE, width=1080, height=132, has-focus=false, has-focusable=false, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=android.widget.FrameLayout$LayoutParams@a54573e, tag=null, root-is-layout-requested=false, has-input-connection=false, x=0.0, y=1788.0} 
|
	at dalvik.system.VMStack.getThreadStackTrace(Native Method)
	at java.lang.Thread.getStackTrace(Thread.java:1538)
	at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:16)
	at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:36)
	at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:106)
	at androidx.test.espresso.ViewInteraction.check(ViewInteraction.java:31)
	at org.mozilla.fenix.ui.robots.NavigationToolbarRobot$Transition.enterURLAndEnterToBrowser(NavigationToolbarRobot.kt:108)
	at org.mozilla.fenix.ui.HistoryTest.visitedUrlHistoryTest(HistoryTest.kt:88)
	at java.lang.reflect.Method.invoke(Native Method)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at androidx.test.internal.runner.junit4.statement.RunBefores.evaluate(RunBefores.java:80)
	at androidx.test.internal.runner.junit4.statement.RunAfters.evaluate(RunAfters.java:61)
	at androidx.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:531)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
	at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:395)
	at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2145)
```
### Build: 
https://github.com/mozilla-mobile/fenix/commit/1502af3d71258333403d537f440807385220c70f

**Comment by sv-ohorvath** [comment]
I logged a bug for this, as it happens on other tests too and suspect it could be a problem in the app, even though I can't manually reproduce it yet: https://github.com/mozilla-mobile/fenix/issues/22316

**Comment by AndiAJ** [comment]
Fixed with #24868
